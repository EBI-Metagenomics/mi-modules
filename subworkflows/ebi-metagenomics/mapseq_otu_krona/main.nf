
include { MAPSEQ             } from '../../../modules/ebi-metagenomics/mapseq/main'
include { MAPSEQ2BIOM        } from '../../../modules/ebi-metagenomics/mapseq2biom/main'
include { KRONA_KTIMPORTTEXT } from '../../../modules/ebi-metagenomics/krona/ktimporttext/main'

workflow MAPSEQ_OTU_KRONA {

    take:
    // TODO nf-core: edit input (take) channels
    ch_fasta    // channel: [ val(meta), [ bam ] ]
    ch_dbs      // channel: [ path(fasta), path(tax), path(otu), path(mscluster), val(label) ]

    main:

    ch_versions = Channel.empty()

    // TODO nf-core: substitute modules here for the modules of your subworkflow

    db_fasta = ch_dbs[0]
    db_tax = ch_dbs[1]
    db_otu = ch_dbs[2]
    db_mscluster = ch_dbs[3]
    db_label = ch_dbs[4]
    

    MAPSEQ(
        ch_fasta,
        tuple(db_fasta, db_tax, db_mscluster)
    )


    // SAMTOOLS_SORT ( ch_bam )
    // ch_versions = ch_versions.mix(SAMTOOLS_SORT.out.versions.first())

    // SAMTOOLS_INDEX ( SAMTOOLS_SORT.out.bam )
    // ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())

    emit:
    // TODO nf-core: edit emitted channels
    mseq      = MAPSEQ.out.mseq          // channel: [ val(meta), [ bam ] ]
    // bai      = SAMTOOLS_INDEX.out.bai          // channel: [ val(meta), [ bai ] ]
    // csi      = SAMTOOLS_INDEX.out.csi          // channel: [ val(meta), [ csi ] ]

    // versions = ch_versions                     // channel: [ versions.yml ]
}

