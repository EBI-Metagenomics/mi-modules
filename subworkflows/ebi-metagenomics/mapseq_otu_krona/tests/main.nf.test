// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core subworkflows test mapseq_otu_krona
nextflow_workflow {

    name "Test Subworkflow MAPSEQ_OTU_KRONA"
    script "../main.nf"
    workflow "MAPSEQ_OTU_KRONA"

    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "subworkflows/mapseq_otu_krona"
    // TODO nf-core: Add tags for all modules used within this subworkflow. Example:
    tag "samtools"
    tag "samtools/sort"
    tag "samtools/index"


    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used
    test("test_mapseq_otu_krona") {

        when {
            workflow {
                """
                // TODO nf-core: define inputs of the workflow here. Example:
                input[0] = [ [ id:'test', single_end:false ], // meta map
                        file("${moduleDir}/tests/subunit_silva_SSU.fasta", checkIfExists: true)
                        ]
                input[1] = [ 
                        file("${moduleDir}/tests/silva_ssu/silva_ssu-20200130_SSU_trimmed.fasta", checkIfExists: true),
                        file("${moduleDir}/tests/silva_ssu/silva_ssu-20200130_SSU_filtered2.txt", checkIfExists: true),
                        file("${moduleDir}/tests/silva_ssu/silva_ssu-20200130_SSU_trimmed.otu", checkIfExists: true),
                        file("${moduleDir}/tests/silva_ssu/silva_ssu-20200130_SSU_trimmed.fasta.mscluster", checkIfExists: true),
                        "SILVA"
                        ]
                """
            }
        }
        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
                //TODO nf-core: Add all required assertions to verify the test output.
            )
        }
    }
}
