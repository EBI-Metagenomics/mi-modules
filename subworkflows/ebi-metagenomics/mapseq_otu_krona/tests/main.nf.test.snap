{
    "test_mapseq_otu_krona": {
        "content": [
            {
                "0": [
                    
                ],
                "mseq": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-17T13:59:04.430544"
    }
}